src/hooks/useGetAuthToken.ts

```diff
@@ -0,0 +1,26 @@
+import { useSearchParams } from "next/navigation";
+import { useEffect, useState } from "react";
+
+const useGetAuthToken = () => {
+  const [token, setToken] = useState<string | null>(null);
+  const searchParams = useSearchParams();
+
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      const localToken = localStorage.getItem("authToken");
+      if (localToken) {
+        setToken(localToken);
+      } else {
+        const authToken = searchParams.get("authToken");
+        if (authToken) {
+          localStorage.setItem("authToken", authToken);
+          setToken(authToken);
+        }
+      }
+    }
+  }, [searchParams]);
```

Comment:
아마 클라이언트에서 렌더링되지 않은 경우에 해당 로직을 실행하면 오류가 발생하기 때문에 `if (typeof window !== "undefined") ` 를 넣으신 것으로 보이는데 맞을까요?
해당 조건은 이미 클라이언트 렌더링 시 동작하는 useEffect 내에서 실행되기 때문에 로그인 시에 window undefined를 확인하는 if문을 제거해도 잘 동작되는 것으로 확인되네요 :)

```suggestion
  useEffect(() => {
    
      const localToken = localStorage.getItem("authToken");
      if (localToken) {
        setToken(localToken);
      } else {
        const authToken = searchParams.get("authToken");
        if (authToken) {
          localStorage.setItem("authToken", authToken);
          setToken(authToken);
        }
      }
  }, [searchParams]);
```